### PR TITLE
Update of the MFT assessment workflow, added several histograms

### PR DESCRIPTION
**Pull request by sarahherrmann**

### CODE REVIEW

File: Detectors/ITSMFT/MFT/assessment/include/MFTAssessment/MFTAssessment.h [hunk 1]
```diff
@@ -32,6 +32,12 @@
 #include "Steer/MCKinematicsReader.h"
 #include <unordered_map>
 #include <vector>
+#include "DataFormatsITSMFT/TopologyDictionary.h"
+#include "MFTTracking/IOUtils.h"
+#include "ReconstructionDataFormats/BaseCluster.h"
+#include "ITSMFTReconstruction/ClustererParam.h"
+#include "DetectorsCommonDataFormats/DetectorNameConf.h"
+#include "MFTBase/GeometryTGeo.h"
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/include/MFTAssessment/MFTAssessment.h]: Hi Sarah, thanks this PR.
Since `"MFTBase/GeometryTGeo.h"` is not needed, please remove all includes and references.

File: Detectors/ITSMFT/MFT/assessment/include/MFTAssessment/MFTAssessment.h [hunk 2]
```diff
@@ -78,11 +84,20 @@ class MFTAssessment
   }
 
  private:
+  o2::itsmft::TopologyDictionary mDictionary; // cluster patterns dictionary
+  //o2::mft::GeometryTGeo* geom;
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/include/MFTAssessment/MFTAssessment.h]: same

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx [hunk 3]
```diff
@@ -12,19 +12,38 @@
 #include "MFTAssessment/MFTAssessment.h"
 #include "Framework/InputSpec.h"
 #include "DetectorsBase/GeometryManager.h"
+#include "MFTBase/GeometryTGeo.h"
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx:15]: same

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx [hunk 4]
```diff
@@ -12,19 +12,38 @@
...
 #include <Framework/InputRecord.h>
 #include <TPaveText.h>
 #include <TLegend.h>
 #include <TStyle.h>
 #include <TFile.h>
+#include <numeric>
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx]: why is this needed?

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx [hunk 5]
```diff
@@ -12,19 +12,38 @@
...
+using o2::itsmft::CompClusterExt;
+o2::itsmft::ChipMappingMFT mMFTMapping;
 
 //__________________________________________________________
 void MFTAssessment::init(bool finalizeAnalysis)
 {
   mFinalizeAnalysis = finalizeAnalysis;
   createHistos();
+  //get geometry
+  o2::base::GeometryManager::loadGeometry("", true);
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx:32]: Not needed, right?

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx [hunk 6]
```diff
@@ -280,6 +346,8 @@ void MFTAssessment::runASyncQC(o2::framework::ProcessingContext& ctx)
     mNOfTracksTime->Fill(seconds, rof.getNEntries());
   }
 
+  std::array<std::array<int, 2>, 5> clsEntriesForRedundancy; //declare it in the .h ?
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx]: No need to declare in the header file. Please remove comment.

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx [hunk 7]
```diff
@@ -288,6 +356,43 @@ void MFTAssessment::runASyncQC(o2::framework::ProcessingContext& ctx)
...
+
+    auto ncls = oneTrack.getNumberOfPoints();
+    auto offset = oneTrack.getExternalClusterIndexOffset();
+
+    for (int icls = 0; icls < ncls; ++icls) //cluster loop
+    {
+
+      auto clsEntry = mMFTTrackClusIdx[offset + icls];
+      auto globalCluster = mMFTClustersGlobal[clsEntry];
+      //auto oneCluster = mMFTClusters[clsEntry];
```
Comment (rpezzi) [inline Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx]: please remove

**Review by rpezzi — CHANGES_REQUESTED** [review]
Some minor changes are necessary. Please see comments.
Thanks.

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx (hunk 4 above)
Comment (sarahherrmann) [reply Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx]: It isn't anymore, I will remove it

File: Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx (hunk 5 above)
Comment (sarahherrmann) [reply Detectors/ITSMFT/MFT/assessment/src/MFTAssessment.cxx:32]: Yes it is needed, because when calling convertToglobalClusters, it needs a loaded geometry

**Review by rpezzi — APPROVED** [review]
Thanks.

**Comment by sarahherrmann** [comment]
Hi ! Just to notify all of you that all checks have passed ! :)
